include/jemalloc/internal/tcache_inlines.h

```diff
@@ -32,13 +32,65 @@ tcache_enabled_set(tsd_t *tsd, bool enabled) {
 	tsd_slow_update(tsd);
 }
 
+static inline unsigned
+thread_nhbins_get(tsd_t *tsd) {
+	return tsd_thread_nhbins_get(tsd);
```

Comment:
Add assert the value is in range.

Reply:
I found that this function is also called when tcache is not enabled. When tcache is set as disabled from the beginning, the value would be TCACHE_NBINS_MAX + 1 as defined in the initializer and fails the assertion. Yet these cases will fall back to arena ultimately with tcache disabled. Thus, the assertion is skipped for these cases. Note that we actually want the value to stay invalid so that when tcache is enabled, it will be properly initialized to the `global_do_not_change_nhbins`. Same for `thread_tcache_max`.

Reply:
NVM, it's the same case discussed above. By initializing things in `tsd_tcache_enabled_data_init`, this is no more a concern.